lib/refs.ts

```diff
@@ -0,0 +1,238 @@
+
+interface $RefsMap {
+  [url: string]: $Ref;
+}
+/**
+ * When you call the resolve method, the value that gets passed to the callback function (or Promise) is a $Refs object. This same object is accessible via the parser.$refs property of $RefParser objects.
+ *
+ * This object is a map of JSON References and their resolved values. It also has several convenient helper methods that make it easy for you to navigate and manipulate the JSON References.
+ *
+ * See https://apitools.dev/json-schema-ref-parser/docs/refs.html
+ */
+export default class $Refs {
+  /**
+   * This property is true if the schema contains any circular references. You may want to check this property before serializing the dereferenced schema as JSON, since JSON.stringify() does not support circular references by default.
+   *
+   * See https://apitools.dev/json-schema-ref-parser/docs/refs.html#circular
+   */
+  public circular: boolean;
+
+  /**
+   * Returns the paths/URLs of all the files in your schema (including the main schema file).
+   *
+   * See https://apitools.dev/json-schema-ref-parser/docs/refs.html#pathstypes
+   *
+   * @param types (optional) Optionally only return certain types of paths ("file", "http", etc.)
+   */
+  paths(...types: string[]): string[] {
+    const paths = getPaths(this._$refs, types);
+    return paths.map((path) => {
+      return getPathFromOs(path.decoded);
+    });
+  }
+
+  /**
+   * Returns a map of paths/URLs and their correspond values.
+   *
+   * See https://apitools.dev/json-schema-ref-parser/docs/refs.html#valuestypes
+   *
+   * @param types (optional) Optionally only return values from certain locations ("file", "http", etc.)
+   */
+  values(...types: string[]): JSONSchema {
+    const $refs = this._$refs;
+    const paths = getPaths($refs, types);
+    return paths.reduce<Record<string, any>>((obj, path) => {
+      obj[getPathFromOs(path.decoded)] = $refs[path.encoded].value;
+      return obj;
+    }, {});
+  }
+
+  /**
+   * Returns `true` if the given path exists in the schema; otherwise, returns `false`
+   *
+   * See https://apitools.dev/json-schema-ref-parser/docs/refs.html#existsref
+   *
+   * @param $ref The JSON Reference path, optionally with a JSON Pointer in the hash
+   */
+  /**
+   * Determines whether the given JSON reference exists.
+   *
+   * @param path - The path being resolved, optionally with a JSON pointer in the hash
+   * @param [options]
+   * @returns
+   */
+  exists(path: string, options: any) {
+    try {
+      this._resolve(path, "", options);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /**
+   * Resolves the given JSON reference and returns the resolved value.
+   *
+   * @param path - The path being resolved, with a JSON pointer in the hash
+   * @param [options]
+   * @returns - Returns the resolved value
+   */
+  get(path: string, options?: $RefParserOptions): JSONSchema4Type | JSONSchema6Type | JSONSchema7Type {
```

Comment:
The return value here might not be JSON Schema. If this library used used on an OpenAPI document for example, it could be any one of these https://spec.openapis.org/oas/v3.1.0#components-object.

Reply:
ahh you're right, sorry.